### ISSUE TITLE
Ability to abort after N errors

### ISSUE THREAD

**Issue by nununo** [rclone/rclone#4670]
#### What is your current rclone version (output from `rclone version`)?
```text
rclone v1.53.1
- os/arch: linux/amd64
- go version: go1.15
```

#### What problem are you are trying to solve?
I'm running `rclone` through `ansible` and, due to the way `ansible` works, I can only see the resulting `stdout` once the command finishes. Sometimes all the files being sync'd have wrong permissions and are therefore skipped due to a permission denied error. But I can only find out about it after the whole command is finished, which can take a long time.

Example:
```text
2020/10/11 00:22:36 ERROR : restic: failed to open directory "restic": open /mnt/caixote/restic: permission denied
2020/10/11 00:22:36 ERROR : 1.txt: Failed to calculate src hash: hash: failed to open: open /mnt/caixote/1.txt: permission denied
```

I would like that `rclone` could be configured to abort after the first error (permission denied or other).

#### How do you think rclone should be changed to solve that?

Introducing flag `--max-errors-before-abort` with a number argument. And keep an error counter which would abort after reaching the provided number.

**Comment by MarcosFelipeBC** [comment]
I would like to try to solve this issue! Could it be valid to hacktoberfest?

**Comment by darthShadow** [comment]
Yeah, that shouldn't be a problem. Ping me when you have submitted the PR so I can mark it for hacktoberfest.

**Comment by MarcosFelipeBC** [comment]
`--max-errors` is going to be an `rclone sync` flag or a global flag?

**Comment by ncw** [comment]
> `--max-errors` is going to be an `rclone sync` flag or a global flag?

It should be a global flag I think.

**Comment by MarcosFelipeBC** [comment]
All right! i think that maybe I wouldn't be able to work on this as soon as I wanted, so if someone else wants to take this no problem!

But if this issue stay open longer, maybe I could take this later
